openmc/surface.py

```diff
@@ -1761,11 +1866,35 @@ def _get_base_coeffs(self):
         f = 2*dx*dz
         g = -2*(dx*dx*x0 + dx*dy*y0 + dx*dz*z0 - cos2)
         h = -2*(dy*dy*y0 + dx*dy*x0 + dy*dz*z0 - cos2)
-        j = -2*(dz*dz*y0 + dx*dz*x0 + dy*dz*y0 - cos2)
+        j = -2*(dz*dz*z0 + dx*dz*x0 + dy*dz*y0 - cos2)
         k = (dx*x0 + dy*y0 + dz*z0)**2 - cos2*(x0*x0 + y0*y0 + z0*z0)
 
         return (a, b, c, d, e, f, g, h, j, k)
 
+    def to_xml_element(self):
+        """Return XML representation of the surface
+
+        Returns
+        -------
+        element : xml.etree.ElementTree.Element
+            XML element containing source data
+
+        """
+        # This method overrides Surface.to_xml_element to generate a Quadric
+        # since the C++ layer doesn't support Cylinders right now
```

Comment:
How about the following approach:
1. Silence ID warnings
2. Create a new, equivalent Quadric instance and set its ID to the current surface's ID
3. Export that